crates/ruff_diagnostics/src/fix.rs

```diff
@@ -4,23 +4,106 @@ use serde::{Deserialize, Serialize};
 
 use crate::edit::Edit;
 
+/// Indicates confidence in the correctness of a suggested fix.
+#[derive(Default, Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[non_exhaustive]
+pub enum Applicability {
```

Comment:
I think this may actually be exhaustive with these members; should I drop the `non_exhaustive`?

Reply:
I think dropping `non_exhaustive` should be fine for now because we don't provide a Rust API, meaning we are in control of all usages and can adopt then when extending the enum.